### PR TITLE
[Feature] Support resize relative position embedding in `SwinTransformer`.

### PR DESCRIPTION
**Pull request by YuanLiuuuuuu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: mmcls/models/backbones/swin_transformer.py [hunk 1]
```diff
@@ -499,3 +501,63 @@ def _prepare_abs_pos_embed(self, state_dict, prefix, *args, **kwargs):
...
+
+    def _prepare_rel_pos_embed(self, state_dict, prefix, *args, **kwargs):
+        all_keys = list(state_dict.keys())
+        state_dict_model = self.state_dict()
+        for key in all_keys:
+            if 'relative_position_bias_table' in key:
+                relative_position_bias_table_pretrained = state_dict[key]
+                relative_position_bias_table_current = state_dict_model[key]
+                L1, nH1 = relative_position_bias_table_pretrained.size()
+                L2, nH2 = relative_position_bias_table_current.size()
```
[the remark this answers is not in the document]
Comment (mzr1996) [reply mmcls/models/backbones/swin_transformer.py:513]: And please add an comment
```python
# Copied from https://github.com/microsoft/unilm/blob/master/beit/semantic_segmentation/mmcv_custom/checkpoint.py#L378
# This interpolation method can avoid degradation of using relative position bias on ADE20k,
# refers to https://github.com/microsoft/unilm/issues/376

File: mmcls/models/backbones/swin_transformer.py [hunk 2]
```diff
@@ -499,3 +501,63 @@ def _prepare_abs_pos_embed(self, state_dict, prefix, *args, **kwargs):
...
+                        z = relative_position_bias_table_pretrained[:, i].view(
+                            src_size, src_size).float().numpy()
+                        f_cubic = interpolate.interp2d(x, y, z, kind='cubic')
+                        all_rel_pos_bias.append(
+                            torch.Tensor(f_cubic(dx,
+                                                 dy)).contiguous().view(-1, 1).
+                            to(relative_position_bias_table_pretrained.device))
+
+                    new_rel_pos_bias = torch.cat(all_rel_pos_bias, dim=-1)
+                    state_dict[key] = new_rel_pos_bias
```
Comment (mzr1996) [inline mmcls/models/backbones/swin_transformer.py:557]: Please add a logger info to inform users about the interpolation.

**Review by mzr1996 — APPROVED** [review]

**Comment by AlenLi817** [comment]
There is an error in test.py for swin backbone checkpoint.pth:
load checkpoint from local path: work_dirs/results/NoPretrain_zuoCfg_ISIC2018_split_80_05_15/swin-tiny_1xb32_zuoCfg_isic2018/best_accuracy_top-1_epoch_81.pth
Traceback (most recent call last):
  File "E:/pyproject/mmclassification-master/alentest.py", line 251, in <module>
    main()
  File "E:/pyproject/mmclassification-master/alentest.py", line 174, in main
    checkpoint = load_checkpoint(model, args.checkpoint, map_location='cpu')
  File "C:\Users\LiAoLun\anaconda3\lib\site-packages\mmcv\runner\checkpoint.py", line 563, in load_checkpoint
    load_state_dict(model, state_dict, strict, logger)
  File "C:\Users\LiAoLun\anaconda3\lib\site-packages\mmcv\runner\checkpoint.py", line 80, in load_state_dict
    load(module)
  File "C:\Users\LiAoLun\anaconda3\lib\site-packages\mmcv\runner\checkpoint.py", line 78, in load
    load(child, prefix + name + '.')
  File "C:\Users\LiAoLun\anaconda3\lib\site-packages\mmcv\runner\checkpoint.py", line 73, in load
    module._load_from_state_dict(state_dict, prefix, local_metadata, True,
  File "E:\pyproject\mmclassification-master\mmcls\models\backbones\swin_transformer.py", line 456, in _load_from_state_dict
    super()._load_from_state_dict(state_dict, prefix, local_metadata,
  File "C:\Users\LiAoLun\anaconda3\lib\site-packages\torch\nn\modules\module.py", line 1372, in _load_from_state_dict
    hook(state_dict, prefix, local_metadata, strict, missing_keys, unexpected_keys, error_msgs)
  File "E:\pyproject\mmclassification-master\mmcls\models\backbones\swin_transformer.py", line 514, in _prepare_relative_position_bias_table
    relative_position_bias_table_current = state_dict_model[key]
KeyError: 'backbone.stages.0.blocks.0.attn.w_msa.relative_position_bias_table'

File: mmcls/models/backbones/swin_transformer.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlenLi817) [reply mmcls/models/backbones/swin_transformer.py:516]: There is an error in test.py for swin backbone checkpoint.pth:
load checkpoint from local path: work_dirs/swin-tiny_1xb32_isic2018/best_accuracy_top-1_epoch_81.pth
Traceback (most recent call last):
File "E:/pyproject/mmclassification-master/alentest.py", line 251, in
main()
File "E:/pyproject/mmclassification-master/alentest.py", line 174, in main
checkpoint = load_checkpoint(model, args.checkpoint, map_location='cpu')
File "C:\Users\LiAoLun\anaconda3\lib\site-packages\mmcv\runner\checkpoint.py", line 563, in load_checkpoint
load_state_dict(model, state_dict, strict, logger)
File "C:\Users\LiAoLun\anaconda3\lib\site-packages\mmcv\runner\checkpoint.py", line 80, in load_state_dict
load(module)
File "C:\Users\LiAoLun\anaconda3\lib\site-packages\mmcv\runner\checkpoint.py", line 78, in load
load(child, prefix + name + '.')
File "C:\Users\LiAoLun\anaconda3\lib\site-packages\mmcv\runner\checkpoint.py", line 73, in load
module._load_from_state_dict(state_dict, prefix, local_metadata, True,
File "E:\pyproject\mmclassification-master\mmcls\models\backbones\swin_transformer.py", line 456, in _load_from_state_dict
super()._load_from_state_dict(state_dict, prefix, local_metadata,
File "C:\Users\LiAoLun\anaconda3\lib\site-packages\torch\nn\modules\module.py", line 1372, in _load_from_state_dict
hook(state_dict, prefix, local_metadata, strict, missing_keys, unexpected_keys, error_msgs)
File "E:\pyproject\mmclassification-master\mmcls\models\backbones\swin_transformer.py", line 514, in _prepare_relative_position_bias_table
relative_position_bias_table_current = state_dict_model[key]
KeyError: 'backbone.stages.0.blocks.0.attn.w_msa.relative_position_bias_table'

**Pull request merged**